tests/unit/test_agent_controller.py

```diff
@@ -52,6 +55,11 @@ def mock_status_callback():
     return AsyncMock()
 
 
+async def send_event_to_controller(controller, event):
+    await controller._on_event(event)
+    await asyncio.sleep(1)
```

Comment:
```suggestion
    await asyncio.sleep(0.1)
```
If we need this (do we?), 1 seems very high